sympy/solvers/tests/test_solveset.py

```diff
@@ -1086,3 +1071,38 @@ def test_issue_8715():
         (Interval.open(-2, oo) - FiniteSet(0))
     assert solveset(eq.subs(x,log(x)), x, S.Reals) == \
         Interval.open(exp(-2), oo) - FiniteSet(1)
+
+def test_issue_9824():
+    assert solveset(sin(x)**2 - 2*sin(x) + 1, x, domain=S.Reals) ==\
+     ImageSet(Lambda(n, 2*n*pi + pi/2), S.Integers)
+
+
+def test_issue_9531_and_9606():
+    # complex solution
+    assert solveset(sinh(x)) == ImageSet(Lambda(n, n*I*pi), S.Integers)
+    # real solution
+    assert solveset(sinh(x), x, S.Reals) == FiniteSet(0)
+
+@SKIP("Comparison Error")
+def test_issue_7914():
```

Comment:
comparison error. I think in LHS `ImageSet(Lambda(n, 2*n*pi - pi/2), S.Integers)` comes but in RHS `ImageSet(Lambda(n, 2*pi*n - pi/2), S.Integers)`. means `2*pi*n ` and `2*n*pi` is making it false.